ios/RNCAsyncStorage.m

```diff
@@ -213,12 +213,32 @@ static void RCTStorageDirectoryCleanupOld(NSString *oldDirectoryPath)
   }
 }
 
+static void _createStorageDirectory(NSString *storageDirectory, NSError **error)
+{
+  [[NSFileManager defaultManager] createDirectoryAtPath:storageDirectory
+                            withIntermediateDirectories:YES
+                                             attributes:nil
+                                                  error:error];
+}
+
 static void RCTStorageDirectoryMigrate(NSString *oldDirectoryPath, NSString *newDirectoryPath, BOOL shouldCleanupOldDirectory)
 {
   NSError *error;
   // Migrate data by copying old storage directory to new storage directory location
   if (![[NSFileManager defaultManager] copyItemAtPath:oldDirectoryPath toPath:newDirectoryPath error:&error]) {
```

Comment:
What if we put a fileExistsAtPath: check before this and if false, call _createStorageDirectory. Then add an "error != nil" to the copyItemAtPath: check to make sure the fileExistsAtPath succeeded and leave the rest of the method unchanged?

Seems a little messy to _try_ to copy and recursively respond to that failure instead of checking if we should create first, putting ourselves in a consistent state in all cases and going from there.

Reply:
hey chris.. i'm down to make this change but i was reading on the apple docs that its better to attempt an operation and respond than to try to predict on how an operation will go based on the file system's current state. https://developer.apple.com/documentation/foundation/nsfilemanager/1415645-fileexistsatpath

since this is a race condition issue and we are the only thing writing to that folder, it'll be okay.
i'll test and try to get something up within today

Reply:
Didn't know about that doc. I think what you're doing is good then!